R/bayesianpredictionperformance.R

```diff
@@ -0,0 +1,259 @@
+    OE.se      = if (options[["measure"]] == "OE" && options[["inputSE"]] != "")                      dataset[, options[["inputSE"]]],
+    OE.cilb    = if (options[["measure"]] == "OE" && sum(unlist(options[["inputCI"]]) != "") == 2)    dataset[, options[["inputCI"]][[1]][1]],
+    OE.ciub    = if (options[["measure"]] == "OE" && sum(unlist(options[["inputCI"]]) != "") == 2)    dataset[, options[["inputCI"]][[1]][2]],
+    N          = if (options[["inputN"]] != "")      dataset[, options[["inputN"]]],
+    O          = if (options[["inputO"]] != "")      dataset[, options[["inputO"]]],
+    E          = if (options[["inputE"]] != "")      dataset[, options[["inputE"]]],
+    slab       = if (options[["inputLabels"]] != "") dataset[, options[["inputLabels"]]],
+    method     = "BAYES",
+    pars       = pars,
+    adapt      = options[["adapt"]],
+    burnin     = options[["burnin"]],
+    sample     = options[["sample"]],
+    n.chains   = options[["chains"]]
+  ),error = function(e)e)
+
+  # error handling
+  if (any(class(fit) %in% c("simpleError", "error"))) {
+    .quitAnalysis(paste0("metamisc package failed with the following error: '", fit[["message"]], "'"))
+  }
+
+  model[["object"]] <- fit
+
+  return()
+}
+.metamiscPriorAndPosteriorPlot <- function(jaspResults, options, dataset, ready) {
+
+  if (!is.null(jaspResults[["priorAndPosteriorPlot"]]))
+    return()
+
+  imgHeight  <- 1000
+  imgWidth   <- 520
+
+  priorAndPosteriorPlot   <- createJaspPlot(title = gettext("Prior and posterior plot"), width = imgWidth, height = imgHeight)
+  priorAndPosteriorPlot$position <- 2.1
+  priorAndPosteriorPlot$dependOn(c("priorAndPosteriorPlot", c(.metamiscDependencies, .metamiscDependenciesBayesian)))
+  jaspResults[["priorAndPosteriorPlot"]] <- priorAndPosteriorPlot
+
+  if (!ready)
+    return()
+
+  p <- try(metamisc::dplot(jaspResults[["model"]][["object"]]))
+
+  if (isTryError(p)) {
+    priorAndPosteriorPlot$setError(.extractErrorMessage(p))
+    return()
+  }
+
+  p <- jaspGraphs::themeJasp(p, sides = "bl") + ggplot2::xlab(gettext("Value")) + ggplot2::ylab(gettext("Density"))
+
+  priorAndPosteriorPlot$plotObject <- p
+  return()
+}
+.metamiscRmPlot                <- function(jaspResults, options, dataset, ready) {
+
+  if (!is.null(jaspResults[["rmPlot"]]))
+    return()
+
+  imgHeight  <- 500
+  imgWidth   <- 300 * options[["chains"]]
+
+  rmPlot   <- createJaspPlot(title = gettext("Running means plot"), width = imgWidth, height = imgHeight)
+  rmPlot$position <- 5
+  rmPlot$dependOn(c("diagnosticsRmPlot", c(.metamiscDependencies, .metamiscDependenciesBayesian)))
+  jaspResults[["rmPlot"]] <- rmPlot
+
+  if (!ready)
+    return()
+
+  p <- try(metamisc::rmplot(jaspResults[["model"]][["object"]]))
+
+  if (isTryError(p)) {
+    rmPlot$setError(.extractErrorMessage(p))
+    return()
+  }
+
+  p <- jaspGraphs::themeJasp(p, sides = "bl") + ggplot2::xlab(gettext("Iteration")) + ggplot2::ylab(gettext("Running mean"))
```

Comment:
`themeJasp` is deprecated in favor of explicitly using `geom_rangeframe` and `themeJaspRaw`.
```suggestion
  p <- p + 
    jaspGraphs::geom_rangeframe() +
    jaspGraphs::themeJaspRaw() + 
    ggplot2::xlab(gettext("Iteration")) + ggplot2::ylab(gettext("Running mean"))
```

Reply:
ok, also propagated to the rest